clientapi/routing/register.go

```diff
@@ -488,10 +488,16 @@ func Register(
 		return *resErr
 	}
 
+	// Extract access token here
+	accessToken, _ := auth.ExtractAccessToken(req)
+
 	// Make sure normal user isn't registering under an exclusive application
 	// service namespace. Skip this check if no app services are registered.
+	// If an access token is provided, ignore this check this is an appservice
+	// request and we will validate in validateApplicationService
```

Comment:
Does the check always happen when there's an access token? i.e. can you now bypass username reservation checks by providing a bogus access token and using dummy login?

Reply:
Further down if we see an access token in the request, we'll treat it like an appservice request. There isn't a way to pass in a token without going through the appservice handler.

Reply:
Unless of course, you pass in a different login type. Hmph.

Reply:
Have made appservice handling entirely special.